projects/plugins/jetpack/modules/widgets/social-icons.php

```diff
@@ -205,25 +205,33 @@ public function widget( $args, $instance ) {
 	 * @return array Updated safe values to be saved.
 	 */
 	public function update( $new_instance, $old_instance ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
-		$instance = array();
+		$instance = $this->defaults;
 
-		$instance['title']     = sanitize_text_field( $new_instance['title'] );
-		$instance['icon-size'] = $this->defaults['icon-size'];
+		if ( ! empty( $new_instance['title'] ) ) {
+			$instance['title'] = sanitize_text_field( $new_instance['title'] );
+		}
 
-		if ( in_array( $new_instance['icon-size'], array( 'small', 'medium', 'large' ), true ) ) {
+		if (
+			! empty( $new_instance['icon-size'] )
+			&& in_array( $new_instance['icon-size'], array( 'small', 'medium', 'large' ), true )
+		) {
 			$instance['icon-size'] = $new_instance['icon-size'];
+		} else {
+			$instance['icon-size'] = $this->defaults['icon-size'];
 		}
 
-		$instance['new-tab'] = isset( $new_instance['new-tab'] ) ? (bool) $new_instance['new-tab'] : false;
-		$instance['icons']   = array();
+		$instance['new-tab'] = isset( $new_instance['new-tab'] ) ? (bool) $new_instance['new-tab'] : $this->defaults['new-tab'];
+		$instance['icons']   = isset( $old_instance['icons'] ) ? $old_instance['icons'] : $this->defaults['icons'];
 
-		foreach ( $new_instance['url-icons'] as $url ) {
-			$url = filter_var( $url, FILTER_SANITIZE_URL );
+		if ( ! empty( $new_instance['url-icons'] ) ) {
+			foreach ( $new_instance['url-icons'] as $i => $url ) {
+				$url = wp_http_validate_url( $url );
```

Comment:
This is a new way of validating the URLs, as `filter_var( $url, FILTER_SANITIZE_URL )` allowed too many false URLs.